client/src/commands/addWalletIdentityCommand.ts

```diff
@@ -82,14 +85,29 @@ export async function addGatewayIdentity(gatewayItem: GatewayTreeItem | FabricGa
 
     if (addIdentityMethod === UserInputUtil.ADD_CERT_KEY_OPTION) {
         // User wants to add an identity by providing a certificate and private key
-        const certKey: {certificatePath: string, privateKeyPath: string} = await UserInputUtil.getCertKey(gatewayRegistryEntry.name);
+        const certKey: {certificatePath: string, privateKeyPath: string} = await UserInputUtil.getCertKey();
         if (!certKey) {
             return Promise.resolve();
         }
         certificatePath = certKey.certificatePath;
         privateKeyPath = certKey.privateKeyPath;
     } else {
         // User wants to add an identity by providing a enrollment id and secret
+        // Ask them what gateway they want to use
+        const chosenEntry: IBlockchainQuickPickItem<FabricGatewayRegistryEntry> = await UserInputUtil.showGatewayQuickPickBox('Choose a gateway to enroll the identity with', true);
+        if (!chosenEntry) {
+            return;
+        }
+        // TODO: gateway needs to be complete before we do this
```

Comment:
wouldn't you just make sure that the gatways shown by the quickpick box were only complete ones?